### ISSUE TITLE
2023 Goals

### ISSUE THREAD

**Issue by radumereuta** [runtimeverification/k#3098]
A few general directions that we would like to follow in developing K in the year 2023:
- IDE integration through the Language Server Protocol #3070
- K REPL - step through the program and semantics and make K, in general, easier to use #2925
  - IDE integration through Debug Adapter Protocol
- KSummarizer/proof navigation tools that would allow for faster and easier-to-understand proving workflows.

**Comment by ehildenb** [comment]
https://docs.google.com/document/d/1NQQGuXiBlRBQMVfIo8eGAAgl3WsUlmVRsx7fJGKOxiA/edit

**Comment by Baltoli** [comment]
# 2023 Merged PRs

I plan to take a look through this list and try to see if there are any major themes that emerge.

https://github.com/runtimeverification/k/pull/3864 Adopt #3862: "Update Bencher CLI version"
https://github.com/runtimeverification/k/pull/3860 hot fix
https://github.com/runtimeverification/k/pull/3858 hot fix (#3857)
https://github.com/runtimeverification/k/pull/3857 hot fix
https://github.com/runtimeverification/k/pull/3855 Enable new TI in kprove
https://github.com/runtimeverification/k/pull/3853 Clean up `{...}<:S` strict cast syntax
https://github.com/runtimeverification/k/pull/3850 Remove `#parseKORE`
https://github.com/runtimeverification/k/pull/3846 Add more outdated warnings
https://github.com/runtimeverification/k/pull/3843 Correct typo in performance tests in CI
https://github.com/runtimeverification/k/pull/3841 Adopt scalafmt for Scala code style
https://github.com/runtimeverification/k/pull/3840 Remove `EAsSet[E]` and `WAsSet[W]` traits in `Transformers.scala` to address IntelliJ error
https://github.com/runtimeverification/k/pull/3834 Various improvements to `POSet`
https://github.com/runtimeverification/k/pull/3828 Move `id` field out of `ProductionReference` into `TypeInferencer`
https://github.com/runtimeverification/k/pull/3825 Move nailgun to lib/kframework/bin
https://github.com/runtimeverification/k/pull/3823 Add `k-which-python` binary in nix
https://github.com/runtimeverification/k/pull/3821 Add proof hint generation flag
https://github.com/runtimeverification/k/pull/3819 Fix llvm backend bins in the k derivation
https://github.com/runtimeverification/k/pull/3816 Address IntelliJ warnings in `Interface.scala`
https://github.com/runtimeverification/k/pull/3814 Add `KItem` subsorts to the disambiguation module when parsing programs
https://github.com/runtimeverification/k/pull/3813 Clean up cast insertion in `TypeInferenceVisitor`
https://github.com/runtimeverification/k/pull/3812 Claim infer cell2
https://github.com/runtimeverification/k/pull/3807 Remove unused `strict` field from `ParseInModule`
https://github.com/runtimeverification/k/pull/3806 Remove mavenix take 2
https://github.com/runtimeverification/k/pull/3805 Revert "Remove mavenix build"
https://github.com/runtimeverification/k/pull/3804 make Pattern implement Comparable and also have an Ordering instance
https://github.com/runtimeverification/k/pull/3802 Remove mavenix build
https://github.com/runtimeverification/k/pull/3798 Add `total` to `#let`-bindings where LHS is a variable
https://github.com/runtimeverification/k/pull/3797 Fix version of Google Java Format used in CI
https://github.com/runtimeverification/k/pull/3796 Fix version of mavenix
https://github.com/runtimeverification/k/pull/3794 Hotfix for release workflow re. macOS / python
https://github.com/runtimeverification/k/pull/3793 Maven plugin for using google-java-format
https://github.com/runtimeverification/k/pull/3792 Update Bencher's version
https://github.com/runtimeverification/k/pull/3790 Run `mvn verify` with appropriate number of cores
https://github.com/runtimeverification/k/pull/3789 Avoid overlap of equations for Bool combinators simplifications
https://github.com/runtimeverification/k/pull/3785 Apply more automatic Java refactoring
https://github.com/runtimeverification/k/pull/3783 Bump LLVM version in packaging to 15
https://github.com/runtimeverification/k/pull/3781 Fix homebrew repo authentication
https://github.com/runtimeverification/k/pull/3779 Revert #3773
https://github.com/runtimeverification/k/pull/3778 Update release.yml
https://github.com/runtimeverification/k/pull/3776 Move homebrew mentions to new repository
https://github.com/runtimeverification/k/pull/3775 update the version of a couple vulnerable dependencies
https://github.com/runtimeverification/k/pull/3774 Update README to address missing dependency
https://github.com/runtimeverification/k/pull/3773 Finally remove PKGBUILD
https://github.com/runtimeverification/k/pull/3772 Fix release cachix action
https://github.com/runtimeverification/k/pull/3771 Revert experimental Homebrew changes
https://github.com/runtimeverification/k/pull/3770 Temporary restore of arch PKGBUILD
https://github.com/runtimeverification/k/pull/3768 Clean up Homebrew after running commands
https://github.com/runtimeverification/k/pull/3767 Fix for Homebrew bottle name on arm64
https://github.com/runtimeverification/k/pull/3766 Install wget in release job
https://github.com/runtimeverification/k/pull/3765 Drop Arch Linux and Debian packaging
https://github.com/runtimeverification/k/pull/3764 Move mac release CI to hosted ARM runners
https://github.com/runtimeverification/k/pull/3761 Update README instructions
https://github.com/runtimeverification/k/pull/3760 Temporary hotfixes for release flakiness
https://github.com/runtimeverification/k/pull/3759 Add cachix binary cache publish step to release
https://github.com/runtimeverification/k/pull/3757 Adopt Google Java style
https://github.com/runtimeverification/k/pull/3755 Apply "silent code cleanup" from IntelliJ
https://github.com/runtimeverification/k/pull/3754 Pin nixpkgs to haskell-backend
https://github.com/runtimeverification/k/pull/3750 Apply Java migration: pattern variables
https://github.com/runtimeverification/k/pull/3749 Revert to nix-community/mavenix in nix build
https://github.com/runtimeverification/k/pull/3748 Update nix install script
https://github.com/runtimeverification/k/pull/3746 Apply Java 17 automatic migration: records
https://github.com/runtimeverification/k/pull/3745 Apply Java 17 automatic migration: enhanced switch
https://github.com/runtimeverification/k/pull/3744 V6.1.0 rc
https://github.com/runtimeverification/k/pull/3743 Check for Java 17 when running K tools
https://github.com/runtimeverification/k/pull/3742 Fix release by installing Calibre 6
https://github.com/runtimeverification/k/pull/3741 Fix for macOS nix / kup breakage
https://github.com/runtimeverification/k/pull/3736 Bump JDK requirement to 17
https://github.com/runtimeverification/k/pull/3734 Slimmer nix runtime closure
https://github.com/runtimeverification/k/pull/3733 Add `lowerBounds` and `upperBounds` methods to `POSet`
https://github.com/runtimeverification/k/pull/3725 Drop mentions of Focal from packaging and install documentation
https://github.com/runtimeverification/k/pull/3724 Rephrase `sed` invocation for macOS
https://github.com/runtimeverification/k/pull/3722 Support for memset hook.
https://github.com/runtimeverification/k/pull/3717 Move PL tutorial link out of main "learn K" section
https://github.com/runtimeverification/k/pull/3714 Move dockerhub image to prebuilt z3
https://github.com/runtimeverification/k/pull/3709 Boost is a runtime dependency
https://github.com/runtimeverification/k/pull/3703 Bump calibre version in release
https://github.com/runtimeverification/k/pull/3702 Use new Z3 image location
https://github.com/runtimeverification/k/pull/3701 Move README from builtin
https://github.com/runtimeverification/k/pull/3700 Add option to print better parse errors
https://github.com/runtimeverification/k/pull/3697 Use `macos-13` everywhere
https://github.com/runtimeverification/k/pull/3696 Point to new `release.nix` upload action
https://github.com/runtimeverification/k/pull/3694 Use prebuilt z3 images
https://github.com/runtimeverification/k/pull/3691 Packaging: `version` path fix
https://github.com/runtimeverification/k/pull/3687 Introduces `--debugger-command` flag to krun
https://github.com/runtimeverification/k/pull/3686 Builing MacOS Package on MacOS 13
https://github.com/runtimeverification/k/pull/3683 Packaging: Move `version` into lib/kframework
https://github.com/runtimeverification/k/pull/3679 Finish all changes to the K frontend, LLVM backend, and Haskell backend relating to multi-ary \and/\or
https://github.com/runtimeverification/k/pull/3678 update scala kore parser to use multi-ary and and or everywhere
https://github.com/runtimeverification/k/pull/3676 change case classes and parser to support multi-ary \and/\or.
https://github.com/runtimeverification/k/pull/3673 Implement new sort inference algorithm
https://github.com/runtimeverification/k/pull/3670 Ordering instance for Sentences (2)
https://github.com/runtimeverification/k/pull/3666 Ensure type inference checks rhsSort <: lhsSort for macro rules
https://github.com/runtimeverification/k/pull/3665 remove build extension for aws s3 that is no longer used
https://github.com/runtimeverification/k/pull/3664 Updating Maven and Testing it
https://github.com/runtimeverification/k/pull/3662 Check if macros were expanded in `ensures` and `requires`
https://github.com/runtimeverification/k/pull/3660 kast --debug-tokens
https://github.com/runtimeverification/k/pull/3659 Add llvm-kompile-matching to K package
https://github.com/runtimeverification/k/pull/3658 Setting $workspace variable in `develop.yml`
https://github.com/runtimeverification/k/pull/3655 Fixing `develop.yml` workflow
https://github.com/runtimeverification/k/pull/3654 Refactor GeneratedTopFormat
https://github.com/runtimeverification/k/pull/3653 Fix output sorting for KPrint
https://github.com/runtimeverification/k/pull/3652 Clean up flake inputs
https://github.com/runtimeverification/k/pull/3650 Finished page title generation
https://github.com/runtimeverification/k/pull/3649 Fix K Version on Debian
https://github.com/runtimeverification/k/pull/3648 Adding arguments to `kprove` flags
https://github.com/runtimeverification/k/pull/3644 Updating `Guice` to 4.0 and deleting temporary injections
https://github.com/runtimeverification/k/pull/3642 Adding arguments to `kast` flags
https://github.com/runtimeverification/k/pull/3641 Update the container's name for correcting installing dependencies.
https://github.com/runtimeverification/k/pull/3639 docs: :memo: Added section 2 docs
https://github.com/runtimeverification/k/pull/3638 Improving  kompile's `--help` and `--help-hidden` messages using descriptors
https://github.com/runtimeverification/k/pull/3637 Installing dependences to bencher execution on develop
https://github.com/runtimeverification/k/pull/3636 Fix parsing left/right-assoc kore terms in kast
https://github.com/runtimeverification/k/pull/3631 Add K-ST
https://github.com/runtimeverification/k/pull/3628 Fix gdate dependence: Use `date -r`  instead of `gdate -d` on macOS to get k version
https://github.com/runtimeverification/k/pull/3627 Adding  CI to release workflow to create a `develop` baseline performance on Bencher
https://github.com/runtimeverification/k/pull/3626 Fix k script to print `--version` correctly in docker container
https://github.com/runtimeverification/k/pull/3623 Update `krun` usage message to contain `[options]`
https://github.com/runtimeverification/k/pull/3622 Revert "Ordering instance for Sentences"
https://github.com/runtimeverification/k/pull/3618 Improving Lesson 1.3 by mentioning longest match while scanning tokens
https://github.com/runtimeverification/k/pull/3617 Update Lesson 1.2 to point to the correct further lesson/link
https://github.com/runtimeverification/k/pull/3615 Fix token constructor error
https://github.com/runtimeverification/k/pull/3611 Add Bencher to track Benchmarks Performance Regression
https://github.com/runtimeverification/k/pull/3609 Testing run nix-flake on self hosted linux machine
https://github.com/runtimeverification/k/pull/3608 Allow some rule-only attributes on sentences that desugar into rules
https://github.com/runtimeverification/k/pull/3603 Adding profiling tests
https://github.com/runtimeverification/k/pull/3602 Make no junk right associative
https://github.com/runtimeverification/k/pull/3600 Remove focal image usage
https://github.com/runtimeverification/k/pull/3597 Internal left/right attributes for c++ unparser
https://github.com/runtimeverification/k/pull/3596 Rule label cannot contain \` or whitespace
https://github.com/runtimeverification/k/pull/3594 Fix kore-integration-tests dev shell
https://github.com/runtimeverification/k/pull/3592 Remove structural attribute
https://github.com/runtimeverification/k/pull/3590 Ordering instance for Sentences
https://github.com/runtimeverification/k/pull/3589 Allowed attributes for sentence types
https://github.com/runtimeverification/k/pull/3588 print no-junk with flat multi-or
https://github.com/runtimeverification/k/pull/3586 add TextToKore.parseModule
https://github.com/runtimeverification/k/pull/3584 Fix Java 19
https://github.com/runtimeverification/k/pull/3581 Changing pull maven repository to cloudrepo
https://github.com/runtimeverification/k/pull/3579 Check parameters on attributes
https://github.com/runtimeverification/k/pull/3578 Remove some old unused attributes
https://github.com/runtimeverification/k/pull/3576 Add `--smt-timeout` flag to `kprove`
https://github.com/runtimeverification/k/pull/3575 Add `--smt-timeout` flag to `krun`
https://github.com/runtimeverification/k/pull/3569 Upload failure tar file to review build issue
https://github.com/runtimeverification/k/pull/3568 Minimize JSON output by dropping unused attributes
https://github.com/runtimeverification/k/pull/3567 Hotfix/maven release push
https://github.com/runtimeverification/k/pull/3562 Add doc for kserver
https://github.com/runtimeverification/k/pull/3561 Checks on main/syntax modules during module exclusion
https://github.com/runtimeverification/k/pull/3560 More performance regressions
https://github.com/runtimeverification/k/pull/3558 Increasing Build timeouts for jammy / focal to match arch / bookworm
https://github.com/runtimeverification/k/pull/3557 Fresh constants for configuration initializers
https://github.com/runtimeverification/k/pull/3554 Revert "AddKoreAttribute as a Frontend Kompiler Pass"
https://github.com/runtimeverification/k/pull/3549 Make K derivation extensible with llvm libraries
https://github.com/runtimeverification/k/pull/3543 update jung dependency
https://github.com/runtimeverification/k/pull/3541 Update release.yml
https://github.com/runtimeverification/k/pull/3529 [ModuleToKore Refactoring] Creatign `GenerateMapCeilAxioms` compiler pass
https://github.com/runtimeverification/k/pull/3525 move integer lemmas out of code excluded by maude
https://github.com/runtimeverification/k/pull/3524 rename MISSING_HOOK_JAVA warning to MISSING_HOOK and make it not hidden
https://github.com/runtimeverification/k/pull/3523 Compare Attributes on Transformers
https://github.com/runtimeverification/k/pull/3522 AddKoreAttribute as a Frontend Kompiler Pass
https://github.com/runtimeverification/k/pull/3516 Set Arch release timeout to be longer
https://github.com/runtimeverification/k/pull/3514 Haskell backend dependency chain updates
https://github.com/runtimeverification/k/pull/3513 Fix release JDK version
https://github.com/runtimeverification/k/pull/3512 Fix release build
https://github.com/runtimeverification/k/pull/3510 Sync hs-backend submodule up manually
https://github.com/runtimeverification/k/pull/3507 JSON creation - use object factory
https://github.com/runtimeverification/k/pull/3506 Upgrade debian version to 12 `bookworm`
https://github.com/runtimeverification/k/pull/3503 .github/workflows/update-deps: sync up booster release tag too
https://github.com/runtimeverification/k/pull/3499 Remove the option to opt out of the attribute whitelist
https://github.com/runtimeverification/k/pull/3497 Correctly escape ASCII values in the range 0-31
https://github.com/runtimeverification/k/pull/3495 add haskell attribute to SET-KORE-SYMBOLIC
https://github.com/runtimeverification/k/pull/3492 delete mounting of host's user and group info
https://github.com/runtimeverification/k/pull/3489 Ensure that non-ASCII characters are escaped when emitting to KORE
https://github.com/runtimeverification/k/pull/3487 Update documentation and tutorials for `group(_)`
https://github.com/runtimeverification/k/pull/3484 Ensure module attributes are checked against the whitelist
https://github.com/runtimeverification/k/pull/3483 simplify passing of environment variables to docker file
https://github.com/runtimeverification/k/pull/3482 Commit flake.lock
https://github.com/runtimeverification/k/pull/3480 update nixpkgs to nixos-23.05
https://github.com/runtimeverification/k/pull/3476 Bumping install-nix-action to v22 and setting mac runners to macOS 13
https://github.com/runtimeverification/k/pull/3475 Introducing the `hs-backend-booster` to K Framework
https://github.com/runtimeverification/k/pull/3469 Remove unit applications in modules
https://github.com/runtimeverification/k/pull/3468 Update booster binary name to `kore-rpc-booster`
https://github.com/runtimeverification/k/pull/3467 Update CheckKLabels to consider --concrete-rules
https://github.com/runtimeverification/k/pull/3466 try a better fix for the issue of creating the maven settings
https://github.com/runtimeverification/k/pull/3464 Add booster to the k derivation
https://github.com/runtimeverification/k/pull/3462 attempt to fix crash in maven deployment
https://github.com/runtimeverification/k/pull/3461 Rangemap
https://github.com/runtimeverification/k/pull/3459 Add `generatedCounter` implicitly to claims
https://github.com/runtimeverification/k/pull/3458 Fix EOF parsing error message crash
https://github.com/runtimeverification/k/pull/3457 Use private runner for release to fix out of space problem
https://github.com/runtimeverification/k/pull/3456 Deploy artifacts to CloudRepo
https://github.com/runtimeverification/k/pull/3455 Make the attribute whitelist opt-out
https://github.com/runtimeverification/k/pull/3454 Improve `--version` implementation on K script
https://github.com/runtimeverification/k/pull/3452 Update all K tests to use `group(_)`
https://github.com/runtimeverification/k/pull/3443 Update attribute whitelist
https://github.com/runtimeverification/k/pull/3441 Emit user-defined groups as group(_)
https://github.com/runtimeverification/k/pull/3438 Throw error message when finding duplicate attributes
https://github.com/runtimeverification/k/pull/3437 Check that sorts are well-formed for left, right, and non-assoc productions
https://github.com/runtimeverification/k/pull/3434 Fix error message when missing sort
https://github.com/runtimeverification/k/pull/3432 Implementing support for pretty printing non-terminals name
https://github.com/runtimeverification/k/pull/3431 Update projects description
https://github.com/runtimeverification/k/pull/3430 Improving kore-print script
https://github.com/runtimeverification/k/pull/3428 Add an integration shell for haskell-backend
https://github.com/runtimeverification/k/pull/3425 Remove all uses of the defunct [transition] attribute
https://github.com/runtimeverification/k/pull/3424 Revert "Remove dead code HaskellKRunOptions"
https://github.com/runtimeverification/k/pull/3423 Try and fix K release
https://github.com/runtimeverification/k/pull/3422 Remove ResolveHeatCoolAttribute -O2/-O3 behavior
https://github.com/runtimeverification/k/pull/3420 Pretty print NonTerminal names in generated txt files
https://github.com/runtimeverification/k/pull/3419 Remove dead code HaskellKRunOptions
https://github.com/runtimeverification/k/pull/3417 Create a hidden category for advanced options
https://github.com/runtimeverification/k/pull/3414 Replacing deprecated option flag at tutorial tests
https://github.com/runtimeverification/k/pull/3412 Restrict `Bytes` token syntax
https://github.com/runtimeverification/k/pull/3409 Update `make clean` to account for a few missed test artifacts
https://github.com/runtimeverification/k/pull/3407 Introduces `--execute-to-branch` to `krun` when using LLVM Backend
https://github.com/runtimeverification/k/pull/3405 V6.0.0 rc
https://github.com/runtimeverification/k/pull/3401 Fixing Kore label munging confusion
https://github.com/runtimeverification/k/pull/3400 Setting `--directory` as deprecated
https://github.com/runtimeverification/k/pull/3397 Add back Maven packages changes
https://github.com/runtimeverification/k/pull/3395 Update missing KLabel checks to stabilize error messages
https://github.com/runtimeverification/k/pull/3391 Make new-format rules without antileft the default
https://github.com/runtimeverification/k/pull/3390 Add attribute whitelist and group(_) attribute
https://github.com/runtimeverification/k/pull/3387 make field in KoreBackend protected instead of private
https://github.com/runtimeverification/k/pull/3382 Add temp-dir option
https://github.com/runtimeverification/k/pull/3379 Fix bottle name again
https://github.com/runtimeverification/k/pull/3378 Add test for no-antileft rules with no explicit SC
https://github.com/runtimeverification/k/pull/3375 Fix GeneratedCounterCell location
https://github.com/runtimeverification/k/pull/3374 Use remote bottle path instead of local
https://github.com/runtimeverification/k/pull/3371 Add UniqueId to all sentences, including generated ones
https://github.com/runtimeverification/k/pull/3369 Revert Maven deployment changes
https://github.com/runtimeverification/k/pull/3366 Fix can't find mvn
https://github.com/runtimeverification/k/pull/3364 Updating mint-llvm Makefile and test from Haskell to LLVM Backend
https://github.com/runtimeverification/k/pull/3362 Working around for cp unable to resolve incontainer tilden expession
https://github.com/runtimeverification/k/pull/3357 Update Matrix links for use in any client [external: #3353]
https://github.com/runtimeverification/k/pull/3356 Add check for manual declaration of isSort
https://github.com/runtimeverification/k/pull/3355 Update version of C tools in docs
https://github.com/runtimeverification/k/pull/3354 Total attribute is allowed only on function symbols
https://github.com/runtimeverification/k/pull/3351 Fixing missing directory in temp container for m2 settings.xml
https://github.com/runtimeverification/k/pull/3349 Display line numbers for all code blocks
https://github.com/runtimeverification/k/pull/3346 Move GenSortPredicateRules into the main pipeline
https://github.com/runtimeverification/k/pull/3342 Add `comm` attributes to `==` and `=/=` operations
https://github.com/runtimeverification/k/pull/3341 Fix incorrect #token construction for Unicode characters
https://github.com/runtimeverification/k/pull/3338 Pt.1 change to push location for mvn packages
https://github.com/runtimeverification/k/pull/3332 Fix github-script v6 breakage
https://github.com/runtimeverification/k/pull/3331 LSP completion: don't insert whitespaces around parens
https://github.com/runtimeverification/k/pull/3330 Update release.yml
https://github.com/runtimeverification/k/pull/3328 add tutorial lesson 2.1
https://github.com/runtimeverification/k/pull/3324 Provide better documentation for concrete and symbolic attributes
https://github.com/runtimeverification/k/pull/3323 Fix artifact names on upload
https://github.com/runtimeverification/k/pull/3319 Fix release tarball upload
https://github.com/runtimeverification/k/pull/3315 Use `gh` to simplify uploading release
https://github.com/runtimeverification/k/pull/3314 Add flag to disable antiLeft priority predicates in KORE
https://github.com/runtimeverification/k/pull/3313 Remove KBMC and other dead code
https://github.com/runtimeverification/k/pull/3312 Set release name earlier in release workflow
https://github.com/runtimeverification/k/pull/3311 Remove `kprove` module properly
https://github.com/runtimeverification/k/pull/3310 Set release title with gh release creation
https://github.com/runtimeverification/k/pull/3309 Correct tar command for producing release
https://github.com/runtimeverification/k/pull/3308 .github/master-push: fix file name
https://github.com/runtimeverification/k/pull/3302 Speculative fix for macOS release timeouts
https://github.com/runtimeverification/k/pull/3299 Switch K master job over to GH Runners
https://github.com/runtimeverification/k/pull/3296 Update timeout
https://github.com/runtimeverification/k/pull/3293 Use longer timeouts in release workflow
https://github.com/runtimeverification/k/pull/3292 Improve lesson 1_08 exercise 2 to point to the source README.md 
https://github.com/runtimeverification/k/pull/3289 Add check that labels provided to --concrete-rules correspond to actual rules
https://github.com/runtimeverification/k/pull/3288 Deleting axiom production when the it doesn't have any terms.
https://github.com/runtimeverification/k/pull/3287 Add warning messages when kompiled def is missing
https://github.com/runtimeverification/k/pull/3282 Add timeouts to release job
https://github.com/runtimeverification/k/pull/3279 Drop `kprove-legacy`
https://github.com/runtimeverification/k/pull/3278 Improve `regression-new/help`
https://github.com/runtimeverification/k/pull/3277 Correct documentation on `simplification` attribute
https://github.com/runtimeverification/k/pull/3275 Re-export kore-rpc server with K
https://github.com/runtimeverification/k/pull/3272 Install C binding headers in Nix
https://github.com/runtimeverification/k/pull/3270 Replacing llvm 14 with llvm 15 as a pre-requirement for K
https://github.com/runtimeverification/k/pull/3263 Add error message for empty Map collections
https://github.com/runtimeverification/k/pull/3261 V5.6.0 rc
https://github.com/runtimeverification/k/pull/3259 Allow bison parsers to be generated as libraries
https://github.com/runtimeverification/k/pull/3251 Delete the java backend!
https://github.com/runtimeverification/k/pull/3249 Write external contributors document
https://github.com/runtimeverification/k/pull/3248 Adopt #3192 for CI
https://github.com/runtimeverification/k/pull/3244 Remove references to java backend from main regression test suite
https://github.com/runtimeverification/k/pull/3243 Update FUN and LOGIK
https://github.com/runtimeverification/k/pull/3242 Update pl-tutorial/1_k/5_types
https://github.com/runtimeverification/k/pull/3239 Workflow/enable community workflow checks
https://github.com/runtimeverification/k/pull/3238 Add g++ to dependencies on debian systems
https://github.com/runtimeverification/k/pull/3230 Automatically use `--no-haskell-binary` in macOS regression tests
https://github.com/runtimeverification/k/pull/3229 Update documentation for new LLDB support
https://github.com/runtimeverification/k/pull/3217 Fixes for mac install of kup
https://github.com/runtimeverification/k/pull/3216 Revert "make K work with Java 19"
https://github.com/runtimeverification/k/pull/3215 Updated README to add fmt as an install prerequisite
https://github.com/runtimeverification/k/pull/3213 disable some modules in the maude backend for the time being
https://github.com/runtimeverification/k/pull/3212 Change invocation of `lldb`
https://github.com/runtimeverification/k/pull/3207 Add flag to enable LLVM backend debugging
https://github.com/runtimeverification/k/pull/3206 make K work with Java 19
https://github.com/runtimeverification/k/pull/3205 Updating cachix references for building ghc
https://github.com/runtimeverification/k/pull/3201 Fix Nix upgrade breaking cachix
https://github.com/runtimeverification/k/pull/3199 Update kool to use llvm backend and fix warnings
https://github.com/runtimeverification/k/pull/3198 Fix loc info for gen config cells
https://github.com/runtimeverification/k/pull/3197 Update deps GH workflow 
https://github.com/runtimeverification/k/pull/3196 Add new flag '--no-pattern' to krun to skip kore-match-disjunction
https://github.com/runtimeverification/k/pull/3195 Tut fix typofix addedlink
https://github.com/runtimeverification/k/pull/3194 Add K-Javalette
https://github.com/runtimeverification/k/pull/3190 .github/: update nix/cachix actions
https://github.com/runtimeverification/k/pull/3186 Exclude nix result dir from checkstyle
https://github.com/runtimeverification/k/pull/3184 Add line numbers to the debugging tutorial
https://github.com/runtimeverification/k/pull/3183 Fix issue with special characters in path
https://github.com/runtimeverification/k/pull/3182 Fix path to k-exercises repo
https://github.com/runtimeverification/k/pull/3181 Update simple to llvm backend
https://github.com/runtimeverification/k/pull/3176 [K MD Parser] Extend TagSelector to Support Line Numbers in Code Blocks
https://github.com/runtimeverification/k/pull/3174 Fix another batch of suggestions from Clarity on tutorial lessons
https://github.com/runtimeverification/k/pull/3173 Add selection range in lsp
https://github.com/runtimeverification/k/pull/3171 Run llvm-kompile using relative paths
https://github.com/runtimeverification/k/pull/3162 Update Imp++ to use LLVM and Haskell backends
https://github.com/runtimeverification/k/pull/3160 Fix some of the issues from Clarity on tutorial lessons
https://github.com/runtimeverification/k/pull/3158 ci: Removed web-related in Dockerfile
https://github.com/runtimeverification/k/pull/3157 Make gh-pages an orphan branch
https://github.com/runtimeverification/k/pull/3153 Fix github set-output files
https://github.com/runtimeverification/k/pull/3151 Add codeowners for version file
https://github.com/runtimeverification/k/pull/3149 Add proper error message
https://github.com/runtimeverification/k/pull/3142 Klsp inner
https://github.com/runtimeverification/k/pull/3136 Depend on libboost-dev for bindings build
https://github.com/runtimeverification/k/pull/3135 Find occurances
https://github.com/runtimeverification/k/pull/3131 Add tabstop completion
https://github.com/runtimeverification/k/pull/3130 Emphasize kup
https://github.com/runtimeverification/k/pull/3128 add maude backend to KompileOptions
https://github.com/runtimeverification/k/pull/3126 Make Set concat partial
https://github.com/runtimeverification/k/pull/3124 Number claims in kprove
https://github.com/runtimeverification/k/pull/3123 Klsp go to definition
https://github.com/runtimeverification/k/pull/3118 Fix copyright check
https://github.com/runtimeverification/k/pull/3117 remove dead code
https://github.com/runtimeverification/k/pull/3116 Address homebrew enthusiastically upgrading the runner environment
https://github.com/runtimeverification/k/pull/3115 Expand on direnv instructions
https://github.com/runtimeverification/k/pull/3113 Install `fmt` on arch
https://github.com/runtimeverification/k/pull/3109 some minor tweaks to existing backends
https://github.com/runtimeverification/k/pull/3107 Build docker images in `${SUBDIR}` context
https://github.com/runtimeverification/k/pull/3104 Lsp
https://github.com/runtimeverification/k/pull/3103 Add KAVM and KPlutus to the active projects webpage
https://github.com/runtimeverification/k/pull/3102 Add FMT as a dependency
https://github.com/runtimeverification/k/pull/3063 Better labels for heat/cool rules of a context
https://github.com/runtimeverification/k/pull/3032 disable comm axiom generation

PR listing script:
```shell
gh pr list                      \
  --repo runtimeverification/k  \
  -s merged                     \
  -L 1000                       \
  --json url,mergedAt,title |
jq -r                           \
  --arg start '2023-01-01'      \
  --arg end '2023-12-31'        \
  '.[] | select(.mergedAt | . >= $start and . <= $end) | select(.title | contains("Update dependency") | not) | .url + " " + .title'
```
